### DISCUSSION TITLE
Is it possible to have the compiler generate the swig interface for you like gm2?

### DISCUSSION THREAD

**Discussion by ghost** [modula3/cm3#1060] [General]
gm2 could generate a swig interface (*.i) for you when you compile a module. You then could use swig to generate bindings to other languages (Python, Perl,...). I know about the existent of swig cm3 but it seems to be the reverse. It generates Modula 3 bindings from C/C++ libraries. I want to be able to use my Modula 3 module from Python instead. Please have a look at:

https://www.nongnu.org/gm2/11/interface_for_python.html

https://www.nongnu.org/gm2/11/producing_a_python_module.html

**Comment by mikanystrom** [comment]
If you don't mind coding in scheme! Garbage collection works! And objects!

https://github.com/modula3/cm3/tree/master/m3-scheme

There may be some bit rot here.

If you're really interested, I can probably find the time to clean it up a
bit.

   Mika
